posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+
+== Jenkins pipeline scripting introduction
+
+It is recommended to adhere to the concept of infrastructure as a code (IaaC) in DevOps pipeline, pipeline scripting-based jenkins jobs are a good example of IaaC.
+
+You can write link:https://www.jenkins.io/doc/book/pipeline/syntax/[Jenkins pipeline code] in one of the following ways:
+
+* Directly in the link:https://www.jenkins.io/doc/book/pipeline/getting-started/[Jenkins UI]. This is a useful way to get started. You would need to take a backup of the Jenkins instance to store the pipeline code because it is part of the Jenkins installation. 
+
+image::/img/blog/pipeline-code-on-jenkins.png[Pipeline code directly on Jenkins ,width=70%,align="center"]
+
+* Jenkinsfile (a plain text file) in Git and mapping it to Jenkins. This is better for ensuring that your configuration is always under version control. If you have specific build and deployment configurations for separate environments, such as dev, staging, and production, you create a separate Jenkinsfile for each environment and store it in that environment specific Git repo  branches. You can either use "Pipeline" type job or "Multibranch pipeline" type job in this case.
+
+image::/img/blog/pipeline-code-on-git.png[Pipeline code on Git ,width=50%,align="center"]
+
+== Building the Jakarta EE & MicroProfile application with Open Liberty on Jenkins
+
+Before writing pipeline script to automate the deployment, you need to make a change in Dockerfile. The standard Dockerfile that you use in general may not enough to do the Jakarta EE & MicroProfile application deployment (to OpenLiberty server) on containerized environment. A sample Openliberty compliant Dockerfile snipped in given, you can configure it as needed (gather the additional files mentioned in this code from open liberty support team).
+
+    FROM icr.io/appcafe/open-liberty:full-java11-openj9-ubi
+    # Add Liberty server configuration including all necessary features
+    COPY --chown=1001:0  server.xml /config/
+    # Modify feature repository (optional)
+    COPY --chown=1001:0 featureUtility.properties /opt/ol/wlp/etc/
+    # This script will add the requested XML snippets to enable Liberty features and grow image to be fit-for-purpose using featureUtility. 
+    # Only available in 'kernel-slim'. The 'full' tag already includes all features for convenience.
+    RUN features.sh
+    # Add interim fixes (optional)
+    #COPY --chown=1001:0  interim-fixes /opt/ol/fixes/
+    .
+    .
+    .
+    .
+    COPY --chown=1001:0 src/main/liberty/config/ /config/
+    COPY --chown=1001:0 ./target/*.war /config/apps/
+    RUN configure.sh
+
+Below given sample pipeline code can perform the code build, packaging and generating docker image and pushing the same to remote docker hub. You can use it by modifying the parameter section (<>)
```

Comment:
Can you change the style of this paragraph, the code, and the parameter explanations to this:

"You configure a Jenkins job by adding build actions. The following sample pipeline code builds the Liberty Java code, packages it into a Docker image, and pushes the Docker image to a remote Docker image repository, such as DockerHub:"

Put the code snippet here. Alter the parameter style/format to match the list below.

"Where:

* git-token is the personal access token you have generated from your Github account.
* remote-docker-image-repository-url is the location of the Docker image repository.
* username is your user name for the Docker image repository.
* password is your password for the Docker image repository.
* docker-image-version is the version number of your Docker image, a unique identifier."

Reply:
Avoiding the line "You configure a Jenkins job by adding build actions." because no "Build actions" in pipeline type job

Reply:
Other changes are done